src/Compilers/CSharp/Test/Semantic/Semantics/InterpolationTests.cs

```diff
@@ -10769,6 +10775,504 @@ public CustomHandler(int literalLength, int formattedCount, ref S s1, ref S s2)
             );
         }
 
+        [Fact]
+        [WorkItem(65470, "https://github.com/dotnet/roslyn/issues/65470")]
+        public void StructReceiver_Lvalue_02()
+        {
+            var code = @"
+using System;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+var l = new StructLogger();
+Console.WriteLine(""logged = {0}"", l._logged);
+l = test(l);
+Console.WriteLine(""logged = {0}"", l._logged);
+
+StructLogger test(StructLogger l)
+{
+    l.Log($""log:{0}"");
+    return l;
+}
+
+internal struct StructLogger
+{
+    public int _logged;
+
+    public void Log([InterpolatedStringHandlerArgument("""")] DummyHandler handler)
+    {
+        _logged++;
+        Console.WriteLine($""StructLogger: "" + handler.GetContent());
+    }
+}
+
+[InterpolatedStringHandler]
+internal ref struct DummyHandler
+{
+    private readonly StringBuilder _builder;
+    public DummyHandler(int literalLength, int formattedCount, StructLogger structLogger)
+    {
+        Console.WriteLine($""Creating DummyHandler"");
+        _builder = new StringBuilder();
+    }
+    public string GetContent() => _builder.ToString();
+
+    public void AppendLiteral(string s) => _builder.Append(s);
+    public void AppendFormatted<T>(T t) => _builder.Append(t);
+}
+";
+
+            var comp = CreateCompilation(new[] { code, InterpolatedStringHandlerAttribute, InterpolatedStringHandlerArgumentAttribute }, options: TestOptions.ReleaseExe);
+            var verifier = CompileAndVerify(comp, expectedOutput: @"
+logged = 0
+Creating DummyHandler
+StructLogger: log:0
+logged = 1
+");
+
+            verifier.VerifyDiagnostics();
```

Comment:
>verifier [](http://example.com/codeflow?start=12&length=8)

Please consider including `verify.VerifyIL("test()", "...");` to verify the expected IL.

Reply:
```StructReceiver_Rvalue``` captures interesting IL changes. IL comparison is brittle. When possible, I prefer testing behavior instead. Which I think is sufficient for the change